nightingale_dispatcher/src/nightingale_dispatcher/task.py

```diff
@@ -1,6 +1,23 @@
 #!/usr/bin/env python3
+import rospy
+from nightingale_ros_bridge import BridgeConfig, RobotStatus, UserInputs
+from nightingale_msgs.srv import InterfaceCall
 
 
 class Task:
-    ERROR = -1
-    SUCCESS = 0
+    def __init__(self):
+        self.ERROR = -1
+        self.SUCCESS = 0
+        self.interface_comms_service_name = BridgeConfig.UPDATE_UI_SERVICE
+
+    def interface_comms_client(self, robot_state):
```

Comment:
im not sure i understand. why should every child of task have this function defined? why should NavigationTask have interface_comms_client defined?

Reply:
I was thinking that every task does not necessarily need to communicate with the user for input but does need to send commands to the screen to tell the screen to change screen based on what the robot is doing. for example the navigate task would send a message to the interface to change to the "driving" screen which does not get back a useful response but is still useful to change the screen. I could make a separate class that inherits from the main "task" class so only tasks that need it inherit it. Though at the moment I think each task does need to update the screen at some point

Reply:
Ben, Felix and I had a discussion about this earlier today. We decided that many of the tasks: Stock, Triage, Handoff, all require some interface calls. It makes sense to have one place where the interface call is defined for each of these tasks, but it should not be in the Task class. The idea is to make a child class for Task called TaskWithInterface (or something like that) that has this interface client. Then the StockTask, TriageTask, and HandoffTask classes all inherit from this TaskWithInterface class instead of Task.

The inhertance structure looks like this:
class Task
class TaskWithInterface(Task)
class StockTask(TaskWithInterface)